compiler/rustc_middle/src/query/mod.rs

```diff
@@ -183,6 +183,14 @@ rustc_queries! {
         separate_provide_extern
     }
 
+    query unsizing_params_for_adt(key: DefId) -> &'tcx rustc_index::bit_set::BitSet<u32>
+    {
+        desc { |tcx|
```

Comment:
```suggestion
    query unsizing_params_for_adt(key: DefId) -> rustc_index::bit_set::BitSet<u32> {
        arena_cache
        desc { |tcx|
```
And we can let the query system do the arena allocation on its own.